src/game/default/g_weapon.c

```diff
@@ -491,6 +509,14 @@ void G_ClientHookThink(g_entity_t *ent) {
 	}
 
 	// send off to the proper sub-function
+
+	if (refire) {
+
+		G_ClientHookCheckFire(ent, true);
+
+		return;
```

Comment:
Not positive, it's when the projectile spawns the velocity seems a bit random (Z seems zero'd sometimes)